junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,544 @@
+              Literal(Constant(mtdSym.nameString)),
+              mkList(annotations)))
+        }
+
+        def mkList(elems: List[Tree]): Tree = {
+          val array = ArrayValue(TypeTree(definitions.ObjectTpe), elems)
+          val castArray = gen.mkCast(array, typeOf[Array[Object]])
+          val wrappedArray = gen.mkMethodCall(
+              definitions.PredefModule,
+              definitions.wrapArrayMethodName(definitions.ObjectTpe),
+              Nil, List(castArray))
+          gen.mkMethodCall(definitions.List_apply, List(wrappedArray))
+        }
+
+        def mkMethodList(tpe: TypeTree)(testMethods: List[MethodSymbol]): Tree =
+          mkList(testMethods.map(defaultMethodMetadata(tpe)))
+
+        val getJUnitMethodRhs = {
+          mkNewInstance(
+              TypeTree(jUnitClassMetadataType),
+              List(
+                mkList(liftAnnotations(clSym)),
+                gen.mkNil,
+                mkMethodList(jUnitMethodMetadataTypeTree)(methods),
+                mlMethods.fold(gen.mkNil)(mkMethodList(jUnitMethodMetadataTypeTree))
+          ))
+        }
+
+        val local = methods.head
+        val getJUnitMetadataSym = local.cloneSymbol
+        getJUnitMetadataSym.withoutAnnotations
+        getJUnitMetadataSym.setName(newTermName("metadata"))
+        getJUnitMetadataSym.setInfo(MethodType(Nil, jUnitClassMetadataType))
+
+        typer.typedDefDef(newDefDef_compat(getJUnitMetadataSym, getJUnitMethodRhs)())
+      }
+
+      private def hasJUnitMethodAnnotation(mtd: MethodSymbol): Boolean =
+        annotationWhiteList.exists(hasAnnotation(mtd, _))
+
+      private def hasAnnotation(mtd: MethodSymbol, tpe: TypeSymbol): Boolean =
+        mtd.annotations.exists(_.atp.typeSymbol == tpe)
+
+      private def mkNewInstance[T: TypeTag](params: List[Tree]): Tree =
+        mkNewInstance(TypeTree(typeOf[T]), params)
+
+      private def mkNewInstance(tpe: TypeTree, params: List[Tree]): Tree =
+        Apply(Select(New(tpe), nme.CONSTRUCTOR), params)
+
+      /* Generate a method in that creates a new instance of the test class, this
+       * method will be located in the bootstrapper class.
+       */
+      private def genNewInstanceDef(classSym: Symbol, bootSymbol: Symbol): DefDef = {
+        val mkNewInstanceDefRhs =
+          mkNewInstance(TypeTree(classSym.typeConstructor), Nil)
+        val mkNewInstanceDefSym = bootSymbol.newMethodSymbol(newTermName("newInstance"))
+        mkNewInstanceDefSym.withoutAnnotations
+        mkNewInstanceDefSym.setInfo(MethodType(Nil, definitions.ObjectTpe))
+
+        typer.typedDefDef(newDefDef_compat(mkNewInstanceDefSym, mkNewInstanceDefRhs)())
+      }
+
+      // Copy of gen.mkClassDef for 2.10 compat
+      private def gen_mkClassDef_compat(mods: Modifiers, name: TypeName,
+          tparams: List[TypeDef], templ: Template): ClassDef = {
+        val isInterface = mods.isTrait &&
+          (templ.body forall treeInfo.isInterfaceMember)
```

Comment:
Prefer the `.forall(...)` notation.
Also either keep this on one line, or put `mods.isTrait &&` on the second line as well.